README.rst

```diff
@@ -54,68 +55,80 @@ code in the `Central System documentation_`.
 
 .. code-block:: python
 
-   import asyncio
-   import logging
-   import websockets
-   from datetime import datetime
-
-   from ocpp.routing import on
-   from ocpp.v20 import ChargePoint as cp
-   from ocpp.v20 import call_result
-
-   logging.basicConfig(level=logging.INFO)
-
-
-   class ChargePoint(cp):
-       @on('BootNotification')
-       def on_boot_notitication(self, charging_station, reason, **kwargs):
-           return call_result.BootNotificationPayload(
-               current_time=datetime.utcnow().isoformat(),
-               interval=10,
-               status='Accepted'
-           )
-
-   async def on_connect(websocket, path):
-       """ For every new charge point that connects, create a ChargePoint instance
-       and start listening for messages.
-
-       """
-       charge_point_id = path.strip('/')
-       cp = ChargePoint(charge_point_id, websocket)
-
-       await cp.start()
-
-
-   async def main():
-       server = await websockets.serve(
-           on_connect,
-           '0.0.0.0',
-           9000,
-           subprotocols=['ocpp2.0']
-       )
-
-       await server.wait_closed()
-
-
-   if __name__ == '__main__':
-       asyncio.run(main())
+    import asyncio
+    import logging
+    import websockets
+    from datetime import datetime
+
+    from ocpp.routing import on
+    from ocpp.v201 import ChargePoint as cp
+    from ocpp.v201 import call_result
+
+    logging.basicConfig(level=logging.INFO)
+
+
+    class ChargePoint(cp):
+        @on('BootNotification')
+        def on_boot_notitication(self, charging_station, reason, **kwargs):
+            return call_result.BootNotificationPayload(
+                current_time=datetime.utcnow().isoformat(),
+                interval=10,
+                status='Accepted'
+            )
+
+
+    async def on_connect(websocket, path):
+        """ For every new charge point that connects, create a ChargePoint
+        instance and start listening for messages.
+        """
+        requested_protocols = websocket.request_headers['Sec-WebSocket-Protocol']
```

Comment:
The bounds of `websocket.request_headers` are not checked. This code will break with a `KeyError` if a client doesn't send the 'Sec-WebSocket-Protocol' header.